src/main/java/uk/gov/companieshouse/officer/delta/processor/tranformer/Transformative.java

```diff
@@ -36,7 +36,11 @@ public interface Transformative<S, T> {
     T factory();
 
     default T transform(S source) throws NonRetryableErrorException {
-        return transform(source, factory());
+        T target = transform(source, factory());
```

Comment:
The return from `factory()` could be a `final` on the class initialised in the constructor. A minor improvement but...